### ISSUE TITLE
PrivateTmp does not work with BindPaths

### ISSUE THREAD

**Issue by kugel-** [systemd/systemd#7272]
### Submission type

 - Bug report

### systemd version the issue has been seen with

234

### Used distribution

Ubuntu 16.04 with systemd 234 via artful repositories

### Background
I'm trying to run IBM Notes in a chroot because it requires ancient software libraries, and is just 32bit. It's becoming too hard to install in on recent hosts.

For this I have prepared a rootfs (in a directory) that contains a Debian Jessie tree with all depedencies and IBM Notes itself.  The rootfs must be read-only for the execution.

Since IBM Notes is a GUI program, the X server socket must be availaible inside the chroot (i.e. I want to bind-mount /tmp/.X11-unix)

The problem shows also with the simplified unit file below. Essentially I can't use PrivateTmp together with BindPaths. using BindReadOnlyPaths doesn't make a difference.

### In case of bug report: Expected behaviour you didn't see

Successful service start

### In case of bug report: Unexpected behaviour you saw

Service fails to start with the following error:
```
Nov 07 13:24:11 tmartitz-pc systemd[1]: Starting IBM Notes in chroot...
Nov 07 13:24:11 tmartitz-pc systemd[9377]: notes-sys.service: Failed at step NAMESPACE spawning /bin/ls: No such file or directory
Nov 07 13:24:11 tmartitz-pc systemd[1]: notes-sys.service: Main process exited, code=exited, status=226/NAMESPACE
Nov 07 13:24:11 tmartitz-pc systemd[1]: Failed to start IBM Notes in chroot.
Nov 07 13:24:11 tmartitz-pc systemd[1]: notes-sys.service: Unit entered failed state.
Nov 07 13:24:11 tmartitz-pc systemd[1]: notes-sys.service: Failed with result 'exit-code'.
```

### In case of bug report: Steps to reproduce the problem

```
[Unit]
Description=IBM Notes in chroot

[Service]
Type=oneshot
ExecStart=/bin/ls -la /tmp
PrivateTmp=yes
BindReadOnlyPaths=/tmp/.X11-unix
```

**Comment by lucaswerkmeister** [comment]
Your unit file works for me on systemd v235 (Arch Linux), as does the following command which should be equivalent if I’m not mistaken:
```sh
systemd-run -qt -p PrivateTmp=yes -p BindReadOnlyPaths=/tmp/.X11-unix /bin/ls -la /tmp
```
Is it possible that the bug was fixed in systemd v235? (Unfortunately, there’s no systemd package in artful-backports yet, so this might be difficult to test for you.)

**Comment by poettering** [comment]
Yeah, this is fixed by a227a4be489333b1b149df124ab284d82397ff2d. Closing.
